### PR TITLE
Feature/gridedit 1302 generate circular clg

### PR DESCRIPTION
**Pull request by BillSenior**

### CODE REVIEW

File: libs/MeshKernel/src/CurvilinearGrid/CurvilinearGridGenerateCircularGrid.cpp [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+            xValues[m] = blockSize * static_cast<double>(m);
+        }
+    }
+    else
+    {
+        // Graded in x-direction
+
+        const int size = parameters.num_columns + 1;
+        int muni = static_cast<int>(static_cast<double>(size) * parameters.uniform_columns_fraction);
+        muni += (size - 1 - muni) % 2;
```
Comment (lucacarniato) [inline libs/MeshKernel/src/CurvilinearGrid/CurvilinearGridGenerateCircularGrid.cpp:34]: use more self explanatory variable names

File: libs/MeshKernel/src/CurvilinearGrid/CurvilinearGridGenerateCircularGrid.cpp [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+                deltaX = blockSize * std::pow(alfm, static_cast<double>(m + 1 - minc - muni));
+            }
+
+            return deltaX;
+        };
+
+        for (int m = 0; m < static_cast<int>(xValues.size()); ++m)
+        {
+            xValue += computeDeltaX(m);
+            xValues[static_cast<UInt>(m)] = xValue;
```
Comment (lucacarniato) [inline libs/MeshKernel/src/CurvilinearGrid/CurvilinearGridGenerateCircularGrid.cpp:63]: is this casting necessary?

File: libs/MeshKernel/src/CurvilinearGrid/CurvilinearGridGenerateCircularGrid.cpp [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+    lin_alg::Matrix<Point> gridPoints(parameters.num_rows + 1, parameters.num_columns + 1);
+
+    const double x0 = parameters.origin_x;
+    const double y0 = parameters.origin_y;
+
+    const double phi = parameters.angle;
+    const double cs = std::cos(phi * constants::conversion::degToRad);
+    const double sn = std::sin(phi * constants::conversion::degToRad);
+
+    std::vector<double> xValues(ComputeXValues(parameters));
```
Comment (lucacarniato) [inline libs/MeshKernel/src/CurvilinearGrid/CurvilinearGridGenerateCircularGrid.cpp:116]: add const

File: libs/MeshKernel/src/CurvilinearGrid/CurvilinearGridGenerateCircularGrid.cpp [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+
+    return radiusValues;
+}
+
+std::vector<double> meshkernel::CurvilinearGridGenerateCircularGrid::ComputeThetaValues(const MakeGridParameters& parameters)
+{
+    const UInt size = static_cast<UInt>(parameters.num_columns + 1);
+    std::vector<double> thetaValues(size);
+
+    double deltaTheta = parameters.block_size_x / parameters.radius_curvature;
```
Comment (lucacarniato) [inline libs/MeshKernel/src/CurvilinearGrid/CurvilinearGridGenerateCircularGrid.cpp:173]: const

File: libs/MeshKernel/src/CurvilinearGrid/CurvilinearGridGenerateCircularGrid.cpp [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+        {
+            yValue += (n < nuni ? blockSize : blockSize * std::pow(alfn, static_cast<double>(n + 1 - nuni)));
+            yValues[static_cast<UInt>(n)] = yValue;
+        }
+    }
+
+    return yValues;
+}
+
+lin_alg::Matrix<meshkernel::Point> meshkernel::CurvilinearGridGenerateCircularGrid::GenerateRectangularGrid(const MakeGridParameters& parameters)
```
Comment (lucacarniato) [inline libs/MeshKernel/src/CurvilinearGrid/CurvilinearGridGenerateCircularGrid.cpp:105]: i belive there is already a function for generating a rectangular grid

**Review by lucacarniato — CHANGES_REQUESTED** [review]

File: libs/MeshKernel/src/CurvilinearGrid/CurvilinearGridGenerateCircularGrid.cpp (hunk 1 above)
Comment (BillSenior) [reply libs/MeshKernel/src/CurvilinearGrid/CurvilinearGridGenerateCircularGrid.cpp:34]: done

File: libs/MeshKernel/src/CurvilinearGrid/CurvilinearGridGenerateCircularGrid.cpp (hunk 3 above)
Comment (BillSenior) [reply libs/MeshKernel/src/CurvilinearGrid/CurvilinearGridGenerateCircularGrid.cpp:116]: done

File: libs/MeshKernel/src/CurvilinearGrid/CurvilinearGridGenerateCircularGrid.cpp (hunk 4 above)
Comment (BillSenior) [reply libs/MeshKernel/src/CurvilinearGrid/CurvilinearGridGenerateCircularGrid.cpp:173]: Neither thetaValues nor deltaTheta can be const

File: libs/MeshKernel/src/CurvilinearGrid/CurvilinearGridGenerateCircularGrid.cpp (hunk 5 above)
Comment (BillSenior) [reply libs/MeshKernel/src/CurvilinearGrid/CurvilinearGridGenerateCircularGrid.cpp:105]: The rectangular grid generated here is, usually slightly, different from the one generated by CurvilinearGridRectangularGrid. DeltaX and deltaY can vary with m and n

Comment (BillSenior) [reply libs/MeshKernel/src/CurvilinearGrid/CurvilinearGridGenerateCircularGrid.cpp:105]: So I have renamed the function to GeneratedGradedRectangularGrid

**Pull request merged**